### ISSUE TITLE
[Feature] Protect against validation/approval while adding/modifying line on document

### ISSUE THREAD

**Issue by mdeweerd** [Dolibarr/dolibarr#7588]
# Feature Request
Warn about or take into account service/product line being edited on orders, invoices, etc when validating/approving the order, invoice, etc.

## Use case
When adding a new line to an order, invoice or other kind of document, I find myself too often making the mistake of clicking on "Valider & Approuver" (Validate&Approve) rather than clicking on "Ajouter" (Add) or "Enregistrer" (Save).
The result is that the line that was being edited as a new or modified entry on the document is lost and therefore the (sometime long) description has to be redone from zéro.

By adding a warning or allowing the "Add or Save" implicitally with the "Validate&Approve", these errors would be avoided and/or the process improved.

## Suggested implementation
- Detect if a new line is being added (some fields [description, price] n00ot empty) or if the line being edited has changed from the original (MD5 or other hash of description, prices, quantity).
- If this detection is "positive", either:
a) Warn that a line is being edited and ask confirmation to proceed with the validation/approval;
b) Similar to a), but add the option to "Add or Save" the line that is being added or changed.

**Comment by atm-maxime** [comment]
IMO, a nicer day of doing this is to stop reloading the page when you click on an action button that displays a confirm box. Confirm box should appear directly, allowing to say "no" and continue creating/editing the card.

**Comment by mdeweerd** [comment]
@atm-maxime :  the changes I suggested can be done in JavaScript;

The reason why the page is being reloaded may be to make sure that the latest lines are listed.  It may happen that lines were added in other views of the page and this more or less ensures that you have the latest version of the document before actually validating.

But the verification that there are lines being edited on the current page can (and should most likely) be done in JavaScript.
